### PR TITLE
Move citations to CITATION.cff format

### PR DESCRIPTION
**Pull request by kboone**
This adds a CITATION.cff file as discussed in #302.

I merged the current list of contributors on the docs/about page with the list of contributors on GitHub, and tried to track down affiliations/orcids for everyone.

### CODE REVIEW

File: docs/about.rst [hunk 1]
```diff
@@ -35,27 +35,9 @@ science (radiative transfer simulations for instance).  Hence
...
-* Caroline Sofiatti
-* Rollin C. Thomas
-* Maria Vincenzi
-* Michael Wood-Vasey
+SNCosmo was originally written by Kyle Barbary in 2012. Since then, over 20 people have
+contributed to its development. A full list of contributors can be found in the
+`CITATION.cff <https://github.com/sncosmo/sncosmo/blob/master/CITATION.cff>`_ file.
+Parsed versions of the author list, including BibTeX entries that can be used to cite
+SNCosmo, are available on `GitHub <https://github.com/sncosmo/sncosmo>`_ and `Zenodo
+<https://doi.org/10.5281/zenodo.592747>`_.
```
Comment (benjaminrose) [inline docs/about.rst:43]: Does Zenodo parse the cff file? In other words, will the authors listing on the Zenodo page update as the cff file updates?

Comment (kboone) [reply docs/about.rst:43]: Yes, Zenodo parses CFF files. See https://citation-file-format.github.io/

Comment (benjaminrose) [reply docs/about.rst:43]: My bad, thanks.

I have read that page several times, but did not pay attention to the Zenodo section. I assumed it just said "hey, they use this format too" and not "They will use the specific file from GitHub." Nice to know, and one less thing to update once we adopt this format.

**Comment by benjaminrose** [comment]
@kboone, I think this PR is good. Ping me on any other PRs you want reviewed before we ship v2.6.0.

**Pull request merged**